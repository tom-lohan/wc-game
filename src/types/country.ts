export type Country = {
	name: string,
	emoji: string,
	group: string,
	ranking: number,
	id: string,
}