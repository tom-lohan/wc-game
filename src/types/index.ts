export type { Country } from './country';
export type { Player } from './player';
export { PageType } from './pageType';