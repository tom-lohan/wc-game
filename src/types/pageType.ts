export enum PageType {
	LANDING_PAGE = 'landing_page',
	ADD_PLAYERS_PAGE = 'add_players_page',
	MAKE_DRAW_PAGE = 'make_draw',
}