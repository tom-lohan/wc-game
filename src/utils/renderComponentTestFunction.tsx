import { configureStore } from '@reduxjs/toolkit'
import { render } from '@testing-library/react'
import React, { ReactNode } from 'react'
import { Provider } from 'react-redux'
import '@testing-library/jest-dom'
import { WCGInitialState } from '../wcgSlice'
import gameReducer from '../wcgSlice'

export const renderComponentTestFunction = (
    children: ReactNode,
    modifiedStore: object = {}
) => {
    const store = {
        wcgame: { ...WCGInitialState },
    }

    const mergedStore = { ...store, ...modifiedStore }

    const mockedStore = configureStore({
        reducer: {
            wcgame: gameReducer,
        },
        preloadedState: mergedStore,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware({ serializableCheck: false }),
    })

    return render(<Provider store={mockedStore}>{children}</Provider>)
}
