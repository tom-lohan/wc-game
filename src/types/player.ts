import { Country } from "./country"

export type Player = {
	name: string,
	email?: string,
	country?: Country,
}