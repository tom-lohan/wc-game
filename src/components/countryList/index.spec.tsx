import { CountryList } from '.'
import { Country } from '../../types'
import { renderComponentTestFunction } from '../../utils/renderComponentTestFunction'

describe('CountryList', () => {
    let mockedCountries: Country[] = []
    beforeEach(() => {
        mockedCountries = [
            {
                name: 'Fakeland',
                emoji: 'ðŸ˜Ž',
                ranking: 1,
                group: 'A',
                id: 'abc',
            },
            {
                name: 'MadeUpLand',
                emoji: 'ðŸ¥¸',
                ranking: 2,
                group: 'A',
                id: 'def',
            },
            {
                name: 'Imaginaria',
                emoji: 'ðŸ¤ ',
                ranking: 3,
                group: 'A',
                id: 'ghi',
            },
        ]
        jest.clearAllMocks()
        jest.restoreAllMocks()
    })

    it('should render correctly', () => {
        const { container } = renderComponentTestFunction(
            <CountryList
                countries={mockedCountries}
                isDraggingFromOtherList={false}
                droppableId="mockedList1"
            />
        )
        expect(container).toMatchSnapshot()
    })
})
