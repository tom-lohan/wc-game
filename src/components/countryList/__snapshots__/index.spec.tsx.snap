// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList should render correctly 1`] = `
<div>
  <div
    class="droppable countryList flex gap-4 flex-wrap "
    data-rbd-droppable-context-id="0"
    data-rbd-droppable-id="mockedList1"
  >
    <div
      class="flex gap-4 flex-wrap "
    >
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="playerCard flex items-center justify-center border-2 bg-white border-black p-3"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="abc"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="abc"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          class="m-0 flex gap-2 justify-center items-center"
        >
          <span>
            Fakeland
          </span>
          <span
            class="text-xl"
          >
            ðŸ˜Ž
          </span>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="playerCard flex items-center justify-center border-2 bg-white border-black p-3"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="def"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="def"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          class="m-0 flex gap-2 justify-center items-center"
        >
          <span>
            MadeUpLand
          </span>
          <span
            class="text-xl"
          >
            ðŸ¥¸
          </span>
        </div>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="playerCard flex items-center justify-center border-2 bg-white border-black p-3"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="ghi"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="ghi"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          class="m-0 flex gap-2 justify-center items-center"
        >
          <span>
            Imaginaria
          </span>
          <span
            class="text-xl"
          >
            ðŸ¤ 
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
